src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java

```diff
@@ -131,6 +133,23 @@ public void testCompareToWithDifferentModuleId() {
                 .isTrue();
     }
 
+    @Test
+    public void testCompareToWithDifferentClass() {
```

Comment:
let keep having comments over pure unit tests to share why it is not possible to cover this by Input files.

Reply:
Technically it is possible, as shown by the change in test above. The issue is it becomes connected with those specific modules and I still don't feel pure utils should really be connected to modules.

Reply:
We try to ban all pure unit tests.
We allow them only as requirements for coverage and only because it is impossible to cover by actual execution.

In your tests, there is null based message - it is not possible, so ok to use pure unit tests with comment above test to explain reason.

But comparison of two messages from checks, should be covered by execution of both Checks or please share why not possible.